manager/snap.go

```diff
@@ -77,7 +77,21 @@ func (snap *Snap) ChangeChannel(pack, channel string) error {
 
 // Install is defined on the PackageManager interface.
 func (snap *Snap) Install(packs ...string) error {
-	out, _, err := RunCommandWithRetry(snap.cmder.InstallCmd(packs...), nil)
+	installError := func(err error, exitCode int, cmdOutput string) (bool, error) {
+		switch exitCode {
+		case 0:
+			return false, nil
+		case 1:
+			return true, nil
+		}
+
+		// If the error is a DNS retryable error then ensure we capture that.
+		if retryable, retryableErr := DNSRetryableError(err, exitCode, cmdOutput); retryableErr != nil || !retryable {
```

Comment:
Why not just push that logic to the interface used by `cmder`?

Reply:
You don't actually want to retry every command, so in order to make this more generic to all commands, you need to pass through the type of command running. This is where it gets a little messy, you essentially need to pass a string of the command context running everywhere and that's unwieldy to me.

Unless you have different retryers depending on the command?

Reply:
I meant the `DNSRetryableError` bit since that's backend-specific.